### PR TITLE
Add roadmap

### PR DESCRIPTION
**Pull request by Patrik-Stas**
Signed-off-by: Patrik Stas <patrik.stas@absa.africa>

### CODE REVIEW

File: roadmap.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+# Phase 2
+### Multitennancy 
+The library was built so that it can power both mobile (usually in the role of holder, prover) and 
+institutional agents (usually in the role of issuer, verifier). The current architecture only supports 
+single tenant architectures. In order to build scalable institutional agents, we need to enable 
+using in multi-tenant context - a single process using `aries-vcx` should be able to 
+manage multiple wallets/agents. 
+
+### Migrating to aries-asgar
```
Comment (airoasis) [inline roadmap.md:38]: There is a typo. aries-asgar --> aries-askar

File: roadmap.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+- deleted legacy code, 
+- migrated CI to Github Actions (building docker, ios, android artifacts)
+- changed library testing approach (favoring integration testing in language wrappers, removing 
+  encrypted mock inputs on rust-level unit testing)
+- thinned language wrappers (wrappers should use functions to access data, rather trying to map out 
+  Rust data structures).
+
+# Phase 1
+This phase is all about decoupling parts of the library into independent modules. We are currently 
+targeting extract 2 independent reusable modules:
```
Comment (mirgee) [inline roadmap.md:23]: `targeting extract` -> `targeting extraction`

File: roadmap.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+exposing asynchronous API. Getting this done might be challenging, as either:
+- `IndySDK` would have to be updated to reuse `aries-askar`,
+- we drop dependency on IndySDK from `aries-vcx` and substitute it with component such as 
+  [indy-utils](https://docs.rs/crate/indy-utils/0.3.2), 
+  [indy-shared-rs](https://github.com/bcgov/indy-shared-rs), 
+  [aries-credx](https://github.com/sovrin-foundation/aries-credx-framework-rs).
+
+
+# Note
+Other tasks might come along way, for example support for anoncreds 2.0, adding support for more
```
Comment (mirgee) [inline roadmap.md:51]: `come along the way` (not nickpicking, can be confused with `come a long way`)

File: README.md [hunk 4]
```diff
@@ -17,6 +17,7 @@
 - This is spin off what has previously been known as LibVCX library. 
 - There's still outstanding work to cleanup code, restructure library so small breaking changes occurs with almost every release.
 - The project currently does not follow semantic versioning. Fow now we are releasing versions `0.x.x`. 
+- See our [roadmap](./roadmap.md)
```
Comment (mirgee) [inline README.md:20]: Missing a dot (this is one of the first things every visitor sees, so let's be really clean and beatiful :))

**Review by mirgee — CHANGES_REQUESTED** [review]
Awesome, really useful doc! 👍 I would have more nitpickes about articles, but not so sure if I (or even a native reader) know what is the correct usage :)

**Review by mirgee — APPROVED** [review]

File: roadmap.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+This phase is all about decoupling parts of the library into independent modules. We are currently 
+targeting extraction of 2 independent reusable modules:
+- mediator agent client
+- Aries protocol state machines
+Once that is done, `aries-vcx` will essentially become opinionated C-Callable glue of independent rust 
+crates, rather than single monolithic opinionated library. Additionally `aries-vcx` itself should 
+become crate consumable from other Rust projects.
+
+# Phase 2
+### Multitennancy
```
Comment (mirgee) [inline roadmap.md:31]: Shoul be spelled `Multitenancy`

**Review by mirgee — CHANGES_REQUESTED** [review]

**Review by mirgee — APPROVED** [review]

**Pull request merged**